src/protocols/bdx/BdxMessages.cpp

```diff
@@ -166,6 +166,32 @@ size_t TransferInit::MessageSize() const
     return WriteToBuffer(emptyBuf).Needed();
 }
 
+void TransferInit::LogMessage() const
+{
+    char fd[kMaxFileDesignatorLen];
+    snprintf(fd, FileDesLength + 1, "%s", FileDesignator);
+
+    switch (MessageType)
+    {
+    case MessageType::SendInit:
+        ChipLogProgress(BDX, "SendInit");
```

Comment:
Can we please have all of the MessageType to-string logic in one place?  This is hard to look at and skips logging of unexpected states, which is the opposite of what we should want.

Reply:
The issue is with the way the BDX message types are currently organized. Specifically, `SendInit` and `ReceiveInit` are both using the `TransferInit` type. I didn't want to change the signatures of the existing functions in the classes that would end up calling `LogMessage` which is why I chose to cache the `MessageType` and determine what BDX message it is here. If you have a better suggestion without touching a lot of existing code, please do share.

Reply:
My suggestion would be to at least give MessageType a to-string facility, since this PR seems to need that frequently.

It sounds like @tcarmelveilleux has more fundamental questions about this design though.

Reply:
I've approved.  It's OK to go ahead without this requested change.